src/storage/src/table/mod.rs

```diff
@@ -208,3 +210,29 @@ fn check_vnode_is_set(vnode: VirtualNode, vnodes: &Bitmap) {
         vnode
     );
 }
+
+pub struct KeyedRow<T: AsRef<[u8]>> {
+    vnode_prefixed_key: TableKey<T>,
+    row: OwnedRow,
+}
+
+impl<T: AsRef<[u8]>> KeyedRow<T> {
+    pub fn new(table_key: TableKey<T>, row: OwnedRow) -> Self {
+        Self {
+            vnode_prefixed_key: table_key,
+            row,
+        }
+    }
+
+    pub fn into_row(self) -> OwnedRow {
+        self.row
+    }
+
+    pub fn vnode(&self) -> &[u8] {
+        self.vnode_prefixed_key.vnode_part()
+    }
+
+    pub fn key(&self) -> &[u8] {
+        self.vnode_prefixed_key.key_part()
+    }
+}
```

Comment:
I'm not quite sure but maybe it's a good idea to `impl Row for KeyedRow`? `Row` trait has a method called `into_owned_row`.

Reply:
After checking all the usage, I think `impl Deref` is more appropriate here.